client/orm/qb/select.go

```
@@ -0,0 +1,272 @@
+// Copyright 2023 beego. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package qb
+
+import (
+	"errors"
+	"github.com/beego/beego/v2/client/orm"
+	"github.com/beego/beego/v2/client/orm/internal/models"
+	"github.com/beego/beego/v2/client/orm/qb/errs"
+
+	"reflect"
+	"strings"
+)
+
+// The Selector is used to construct a SELECT statement
+type Selector[T any] struct {
+	sb        strings.Builder
+	args      []any
+	orderBy   []Column
+	where     []Predicate
+	offset    int
+	limit     int
+	model     *models.ModelInfo
+	columns   []Selectable
+	tableName string
+
+	cache *models.ModelCache
+	db    orm.Ormer
+}
+
+func NewSelector[T any](cache *models.ModelCache) *Selector[T] {
+	return &Selector[T]{
+		cache: cache,
+	}
+}
+func (s *Selector[T]) Build() (*Query, error) {
+	var (
+		t   T
+		err error
+	)
+
+	s.model, _ = s.cache.GetByMd(&t)
```

Do not ignore this error